### DISCUSSION TITLE
plugins for embedding html in html exports?

### DISCUSSION THREAD

**Discussion by mid-kid** [zim-desktop-wiki/zim-desktop-wiki#2618] [Ideas]
I'm looking to embed a snippet of html, for prettier formatting of some specific content, into the output of my pages. I know this flies in the face of the WYSIWYG nature of Zim, but the content I want to embed is viewable by clicking on the link in my notebook otherwise, and it's not particularly important for editing the rest of the article.

Does there happen to be a plugin that already does this? Or maybe some builtin method I never found yet?

**Comment by jaap-karssenberg** [comment]
There is no such plugin yet. Probably you could copy the source view plugin
and modify the export code of a code block to achieve this.

Or better, import and inherit from relevant objects in the source view
plugin.

Regards,

Jaap

Op vr 31 mei 2024 13:27 schreef mid-kid ***@***.***>:

**Reply by maclomhair** [reply]
I've written such a plugin quickly once, coincidentally based on the source view plugin as you proposed! No changes to the zim codebase were needed. If I can get to it, I'll upload it and people can use it as an external plugin for now. I don't think it'd be ready quality-wise for inclusion in zim itself...

**Comment by maclomhair** [comment]
Ok so here's the plugin for raw HTML elements in zim pages which I use to add specific layout to my websites made with zim:

https://codeberg.org/maclomhair/zim-plugin-rawhtml

It's very much a proof of concept but it gets the job done. I hope it's useful to you!

**Comment by jaap-karssenberg** [comment]
Yes, this is an easy plugin to make based on the source view plugin. Would
like to have such a plugin in the core release as well.

My only requirement would be to somehow make the two kinds of coffee blocks
visually distinct in the Zim page. Maybe a border color or something?

Op vr 13 sep. 2024 08:52 schreef maclomhair ***@***.***>:

**Reply by maclomhair** [reply]
I now tried a frame with a label to make it clear what is happening with such blocks:

![Screenshot_2025-06-08_11-49-45](https://github.com/user-attachments/assets/2d445c82-750a-4785-b1ad-7d6dcada0cf2)

It's currently in a [separate repo](https://codeberg.org/maclomhair/zim-plugin-rawhtml) but if this is ok, I can submit a pull request :smile:
